### PR TITLE
[DS-3808] Return 204 Status when return json object is empty

### PR DESCRIPTION
**Pull request by terrywbrady**
https://jira.duraspace.org/browse/DS-3808

### CODE REVIEW

File: dspace-spring-rest/src/test/java/org/dspace/app/rest/BitstreamRestRepositoryIT.java [hunk 1]
```diff
@@ -347,7 +347,8 @@ public void findOneBitstreamRelsTest() throws Exception {
 
 
         getClient().perform(get("/api/core/bitstreams/" + bitstream.getID() + "/self"))
-                   .andExpect(status().isOk())
+                   //Does this status make sense?  This is what is being returned.
```
Comment (abollini) [inline dspace-spring-rest/src/test/java/org/dspace/app/rest/BitstreamRestRepositoryIT.java]: no this has no sense... this endpoint doesn't exist at all so a 404 should be returned. Nevertheless it is a separate issue. Can you remove the check of the /self and fire a bug report?

File: dspace-spring-rest/src/test/java/org/dspace/app/rest/SubmissionDefinitionsControllerIT.java [hunk 2]
```diff
@@ -95,8 +95,8 @@ public void findByCollection() throws Exception {
     public void findCollections() throws Exception {
         //Match only that a section exists with a submission configuration behind
         getClient().perform(get("/api/config/submissiondefinitions/traditional/collections"))
-                   //The status has to be 200 OK
-                   .andExpect(status().isOk());
+                   //Does this status change make sense?
```
Comment (abollini) [inline dspace-spring-rest/src/test/java/org/dspace/app/rest/SubmissionDefinitionsControllerIT.java]: yes and no, acording to the rest contract https://github.com/DSpace/Rest7Contract/blob/master/submissiondefinitions.md#collections
as there is no explicit mapping toward the traditional submissionform an empty list is returned...
I'm not sure if no content is the right return here and more in general where list are expected... maybe a zero size page is a better option?

**Comment by terrywbrady** [comment]
I created the following Jira tickets to capture the outstanding issues related to empty return objects.

- https://jira.duraspace.org/browse/DS-3916
- https://jira.duraspace.org/browse/DS-3915

Hopefully, these changes will be an improvement in the interim.

**Review by abollini — APPROVED** [review]
tested it works as expected

**Pull request merged**